store/store.go

```diff
@@ -1109,21 +1103,31 @@ func (s *Store) Download(name string, downloadInfo *snap.DownloadInfo, pbar prog
 		url = downloadInfo.DownloadURL
 	}
 
-	if err := download(name, url, user, s, w, pbar); err != nil {
-		return "", err
+	sha3_384, err := download(name, url, user, s, w, pbar)
+	if err != nil {
+		return err
+	}
+
+	if downloadInfo.Sha3_384 != "" && sha3_384 != downloadInfo.Sha3_384 {
+		// FIXME: find a better error message
+		return fmt.Errorf("hashsum mismatch for %s: got %s but expected %s", w.Name(), sha3_384, downloadInfo.Sha3_384)
```

Comment:
`("sha3-384 mismatch downloading %s: got %s but expected %s", w.Name(), hash, downloadInfo.Sha3_384)`

Note that this is different from the suggestion from the other PR, as here we can use the actual filename since it'll look nice (yay!).